k8s-plugin/src/main/kotlin/com/amazon/spinnaker/keel/k8s/model/K8sManifests.kt

```diff
@@ -91,8 +98,8 @@ data class K8sCredentialManifest(
     override var kind: String?,
     override var metadata: Map<String, Any?>,
     override var spec: K8sSpec?,
-    override var data: K8sData? = null,
+    override var data: Map<String, Any?>? = null,
     override var status: Status? = null,
 ): K8sManifest(apiVersion, kind, metadata, spec, data, status) {
-    override fun name(): String = "${metadata[TYPE]}-${data?.account}"
+    override fun name(): String = "${metadata[TYPE]}-${metadata[CLOUDDRIVER_ACCOUNT]}"
```

Comment:
why do we need to change our own spec to support config map? account name under `data` appears to be nicer. even moving `git` under data is probably ok. any particular reason for not doing the following?

```suggestion
    override fun name(): String = "${metadata[TYPE]}-${data?.get("account")}"
```

Reply:
From end user's perspective, I think it's nicer to have users put information under one location, `metadata`, than have spread them out into metadata and data fields. Data fields are encoded and it's kind of pain to figure out where a particular secret came from. I think metadata is a better place to indicate users which clouddrvier account this secret  corresponds to. You can't really put your own fields in data through this anyways.

Reply:
hmmm, not sure I am following. what we have here under `K8sCredentialManifest` is what the users of MD see when they write their delivery config. It doesn't have much to do with the encoding that gets passed along as a secret. We take one `K8sCredentialManifest` and convert it to another `K8sCredentialManifest` [here](https://github.com/nimakaviani/managed-delivery-k8s-plugin/blob/b4404859614388e826080e0b139d48590da2eecd/k8s-plugin/src/main/kotlin/com/amazon/spinnaker/keel/k8s/resolver/CredentialsResourceHandler.kt#L80-L89) but they are independent of one another.

I agree with the part that splitting the account type and the account name  between metadata and data section is probably not a good idea, but having both under metadata is a bit weird because what gets passed is actually account data IMO.

Reply:
For example, when you have to troubleshoot an issue with a secret created by this handler, wouldn't it be nicer to be able to just see where it comes from by just running `kubectl get secret example` and look at metadata? Instead of doing something like `{{index .data "username" | base64decode}}`?

Reply:
Per our offline conversation: 
1. We will not change the spec
2. We will add annotations to resulting secrets to indicate where this secret was derived from. This however needs to be done after we implement our own annotation process.

Reply:
sounds good, thanks!